packages/blade/src/components/Checkbox/Checkbox.tsx

```diff
@@ -0,0 +1,204 @@
+import React from 'react';
+import { isEmpty } from '../../utils';
+import { useCheckboxGroupContext } from './CheckboxGroup/CheckboxGroupContext';
+import { CheckboxHintText } from './CheckboxHintText';
+import { CheckboxIcon } from './CheckboxIcon';
+import { CheckboxInput } from './CheckboxInput';
+import { CheckboxLabel } from './CheckboxLabel';
+import { CheckboxLabelText } from './CheckboxLabelText';
+import { useCheckbox } from './useCheckbox';
+import { Wrapper } from './Wrapper';
+
+type CheckboxProps = {
+  /**
+   * If `true`, sets the checkbox will be checked.
+   * Use `onChange` to update its value
+   *
+   * @default false
+   */
+  isChecked?: boolean;
+  /**
+   * If `true`, the checkbox will be initially checked.
+   *
+   * @default false
+   */
+  defaultChecked?: boolean;
+  /**
+   * The callback invoked when the checked state of the `Checkbox` changes.
+   */
+  onChange?: (isChecked: boolean) => void;
+  /**
+   * Sets the label text of the checkbox
+   */
+  children: string;
+  /**
+   * Help text of the checkbox
+   */
+  helpText?: string;
+  /**
+   * Error text of the checkbox
+   *
+   * Renders when `validationState` is set to 'error'
+   *
+   * Overrides helpText
+   */
+  errorText?: string;
+  /**
+   * If `true`, the checkbox will be indeterminate.
+   * This does not modify the isChecked property.
+   *
+   * @default false
+   */
+  isIndeterminate?: boolean;
+  /**
+   * The name of the input field in a checkbox
+   * (Useful for form submission).
+   */
+  name?: string;
+  /**
+   * The value to be used in the checkbox input.
+   * This is the value that will be returned on form submission.
+   */
+  value?: string;
+  /**
+   * If `true`, the checkbox will be disabled
+   *
+   * @default false
+   */
+  isDisabled?: boolean;
+  /**
+   * If `true`, the checkbox input is marked as required,
+   * and `required` attribute will be added
+   *
+   * @default false
+   */
+  isRequired?: boolean;
+  /**
+   * If `error`, the checkbox input is marked as invalid,
+   * and `invalid` attribute will be added
+   *
+   * @default undefined
+   */
+  validationState?: 'error' | undefined;
+};
+
+const Checkbox = ({
+  defaultChecked,
+  validationState,
+  isChecked,
+  isDisabled,
+  isIndeterminate,
+  isRequired,
+  name,
+  onChange,
+  value,
+  children,
+  helpText,
+  errorText,
+}: CheckboxProps): React.ReactElement => {
+  const groupProps = useCheckboxGroupContext();
+
+  // ban controlling checkbox manually while inside group
+  const hasDefaultChecked = !isUndefined(defaultChecked);
+  const hasIsChecked = !isUndefined(isChecked);
+  const hasOnChange = !isUndefined(onChange);
+  if ((hasDefaultChecked || hasIsChecked || hasOnChange) && !isEmpty(groupProps)) {
+    const props = [
+      hasDefaultChecked ? 'defaultChecked' : undefined,
+      hasIsChecked ? 'isChecked' : undefined,
+      hasOnChange ? 'onChange' : undefined,
+    ].filter(Boolean);
+
+    throw new Error(
```

Comment:
not a blocker, long term todo: but we shall create a tiny logger function that will format the messages for all the blade things, like prefixing the `[Blade component name]` etc. so we don't have to do this manually for all the things

Reply:
Yes exactly what I was also thinking. 

I was thinking that we should also have a constants for all our errors which we should reference like: CHECKBOX_LABEL_ERROR. That way we can reuse those in the tests too.

Reply:
Sure make sense. Then I would rather snapshot the Error object than manually writing the error msg again.

Reply:
Opened an issue we can track it later: https://github.com/razorpay/blade/issues/580